### PR TITLE
feat: integrate idir lookup in role assignment page

### PR DESCRIPTION
**Pull request by junminahn**

### CODE REVIEW

File: app/components/GenericModal.tsx [hunk 1]
```diff
@@ -118,6 +122,10 @@ const GenericModal = (
     window.location.hash = '#';
   };
 
+  if (typeof children === 'function') {
```
Comment (thegentlemanphysicist) [inline app/components/GenericModal.tsx:125]: What is this doing? Is there a reason why the children can be both a function and an object?

File: app/html-components/Table.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ const Table = styled.table<{ variant?: string; readOnly?: boolean }>`
   text-align: left;
   border-collapse: separate;
   border-spacing: 0 ${TABLE_ROW_SPACING}px;
-  table-layout: fixed;
+  // table-layout: fixed;
```
Comment (thegentlemanphysicist) [inline app/html-components/Table.tsx:11]: should this be deleted?

File: app/page-partials/my-dashboard/UserRoles.tsx [hunk 3]
```diff
@@ -216,7 +290,32 @@ const UserRoles = ({ selectedRequest, alert }: Props) => {
           <td>{row.firstName}</td>
           <td>{row.lastName}</td>
           <td>{row.email}</td>
-          <td>{row.username}</td>
+          <td>
+            <td>
```
Comment (thegentlemanphysicist) [inline app/page-partials/my-dashboard/UserRoles.tsx:294]: Is there supposed to be nested < td >'s here?

File: lambda/app/src/bceid-webservice-proxy/idir.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  console.log({
+    guid: result.guid,
+    userId: result.userId,
+    email: result.contact.email,
+    firstName: result.individualIdentity.name.firstname,
+    lastName: result.individualIdentity.name.surname,
+    displayName: result.displayName,
+  });
+
+  const aa = await Promise.all(
```
Comment (thegentlemanphysicist) [inline lambda/app/src/bceid-webservice-proxy/idir.ts:36]: Is it possible to get a more descriptive name for this constant? Or is this a standard name for a lambda function constant?

**Review by thegentlemanphysicist — COMMENTED** [review]
Left some comments and questions

File: lambda/app/src/bceid-webservice-proxy/idir.ts (hunk 4 above)
Comment (junmin-bcgov) [reply lambda/app/src/bceid-webservice-proxy/idir.ts]: right, forgot to remove the constant; I removed the variable.

File: app/page-partials/my-dashboard/UserRoles.tsx (hunk 3 above)
Comment (junmin-bcgov) [reply app/page-partials/my-dashboard/UserRoles.tsx]: right, removed the unnecessary `td`

File: app/html-components/Table.tsx (hunk 2 above)
Comment (junmin-bcgov) [reply app/html-components/Table.tsx]: removed the commented css line

File: app/components/GenericModal.tsx (hunk 1 above)
Comment (junmin-bcgov) [reply app/components/GenericModal.tsx:125]: there is a usecase to pass function as the children to access to the context information;
https://github.com/bcgov/sso-requests/blob/49c6e83a819b356e8f3f3c914046d5e8853e2e9b/app/page-partials/my-dashboard/UserRoles.tsx#L499

Comment (thegentlemanphysicist) [reply app/components/GenericModal.tsx:125]: Cool thanks for clearing up :)

**Review by thegentlemanphysicist — APPROVED** [review]
Thanks for addressing my comments. :)

**Pull request merged**